### DISCUSSION TITLE
Generating mesh for a unit cell in MOOSE

### DISCUSSION THREAD

**Discussion by yfche** [idaholab/moose#23646] [Q&A Meshing] [answered]
The attached mesh is generated in cubit, which contains six 1/3 cylinders, a central hole, and a hexagon excluding the 1/3 cylinders. I wonder if this geometry can be directly generated and meshed in the MOOSE reactor module. Thanks.
![unit_cell](https://user-images.githubusercontent.com/24327462/223553206-6e0d51e8-e70e-4cdc-90eb-a2441e86455d.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

I think we can do this geometry.
Overall it will be:
- a hexagonconcentriccirclegenerator for the center with a big circle
- same generator for the 6 other positions
- an assembly generator to tie them all together
- an hexagon trimmer with the peripheral option to cut on the circles of the 6 external positions

Guillaume

**Reply by yfche** [reply]
I found that adding "quad_center_elements = true"  causes Segmentation fault: 11, as below. Setting it to false generates the mesh fine. Any ideas why setting the quad_center_elements crashes my code? Thanks.

    [hex_1]
        type = PolygonConcentricCircleMeshGenerator
        num_sides = 6
        num_sectors_per_side = '10 10 10 10 10 10'
        # background_intervals = 10
        # background_block_ids = 20
        # background_block_names = background
        polygon_size = 3.0
        polygon_size_style = radius
        preserve_volumes = off # on
        flat_side_up = true
        quad_center_elements = true
      []

**Reply by GiudGiud** [reply]
@miaoyinb if you could please have a look

**Reply by yfche** [reply]
In addition, the central hole is not of the same size as the 6 external circles. In this case, can we still use the hexagonal assembly to tie these circles together? If not, is my understanding correct that we may not use an hexagon trimmer to cut on the 6 external circles?

**Reply by GiudGiud** [reply]
yes. The central circle is not determining the boundary mesh

For the trimmer there may be some limitations on when you can use it, but I think there's a workaround. Please let us know once you reach there

**Reply by miaoyinb** [reply]
> I found that adding "quad_center_elements = true" causes Segmentation fault: 11, as below. Setting it to false generates the mesh fine. Any ideas why setting the quad_center_elements crashes my code? Thanks.
> 
> ```
> [hex_1]
>     type = PolygonConcentricCircleMeshGenerator
>     num_sides = 6
>     num_sectors_per_side = '10 10 10 10 10 10'
>     # background_intervals = 10
>     # background_block_ids = 20
>     # background_block_names = background
>     polygon_size = 3.0
>     polygon_size_style = radius
>     preserve_volumes = off # on
>     flat_side_up = true
>     quad_center_elements = true
>   []
> ```

This should be a bug. I will fix it. 
It should work fine if you uncomment `# background_block_ids = 20` before the fix.

**Reply by miaoyinb** [reply]
#23652 Thanks for reporting the bug.

**Reply by yfche** [reply]
Uncommenting `# background_block_ids = 20 `works. Thanks for providing the workaround.

**Reply by miaoyinb** [reply]
The fix for the bug has been merged. Your original input should also work with the most updated MOOSE next branch.

**Reply by GiudGiud** [reply]
use the BlockDeletionGenerator
